### PR TITLE
fix generateId

### PR DESCRIPTION
**Pull request by sbmaxx**
#119

### CODE REVIEW

**Comment by sbmaxx** [comment]
@veged @indutny take a look, please

File: lib/bemhtml/runtime/context.js [hunk 1]
```diff
@@ -48,9 +46,7 @@ Context.prototype.isLast = function isLast() {
 };
 
 Context.prototype.generateId = function generateId() {
-  if (this._uniq === null)
-    this._uniq = utils.getUniq();
-  return this._uniq;
+  return utils.getUniq();
```
Comment (indutny) [inline lib/bemhtml/runtime/context.js]: To be compatible with old bem-xjst, it should actually generate id based on current context. Is it ok to generate random here, @veged ?

Comment (zxqfox) [inline lib/bemhtml/runtime/context.js]: +1 for `this.identify(this.ctx)` analogue.

Comment (zxqfox) [inline lib/bemhtml/runtime/context.js]: @sbmaxx try this:
```diff
$ git diff
diff --git a/lib/bemhtml/runtime/context.js b/lib/bemhtml/runtime/context.js
index 3e275ec..d209f3e 100644
--- a/lib/bemhtml/runtime/context.js
+++ b/lib/bemhtml/runtime/context.js
@@ -46,7 +46,7 @@ Context.prototype.isLast = function isLast() {
 };

 Context.prototype.generateId = function generateId() {
-  return utils.getUniq();
+  return utils.identify(this.ctx);
 };

 Context.prototype.reapply = function reapply(ctx) {
```

Comment (veged) [inline lib/bemhtml/runtime/context.js]: yes, it must be same for same node

**Comment by sbmaxx** [comment]
@veged @zxqfox fixed

File: test/runtime-test.js [hunk 2]
```diff
@@ -625,6 +625,48 @@ describe('BEMHTML compiler/Runtime', function() {
...
+            tag()(''),
+            content()(function() {
+              return this.generateId();
+            })
+          )
+        })
+      });
+
+      it('starts with uniq', function() {
+
```
Comment (zxqfox) [inline test/runtime-test.js]: Here you have a spacer

File: test/runtime-test.js [hunk 3]
```diff
@@ -625,6 +625,48 @@ describe('BEMHTML compiler/Runtime', function() {
...
+        var str = template.apply({ block: 'b1' });
+        assert(str.indexOf('uniq') === 0);
+
+      });
+
+      it('should be unique in different applies', function() {
+        assert.notEqual(
```
Comment (zxqfox) [inline test/runtime-test.js]: And here you don't

File: test/runtime-test.js [hunk 4]
```diff
@@ -625,6 +625,48 @@ describe('BEMHTML compiler/Runtime', function() {
...
+          template.apply({ block: 'b1' }),
+          template.apply({ block: 'b1' })
+        );
+      });
+
+      it('should be unique in one apply', function() {
+
```
Comment (zxqfox) [inline test/runtime-test.js]: And again ;-)

**Comment by zxqfox** [comment]
I think we missing one more test case:
```
block('b')(
  tag()(''),
  content()(function(){
    return [this.generateId(), sep, this.generateId()];
  })
);
// and
var res = t.apply({block: 'b'}).split(sep);
assert.equal(res[0], res[1]);
```

**Comment by sbmaxx** [comment]
@zxqfox thx for review ;)

**Pull request merged**